index.js

```diff
@@ -0,0 +1,166 @@
+/* Ethical ad publisher JavaScript client
+ *
+ * Loads placement from Ethical Ad decision API. Searches for elements with
+ * `ethical-ad` data binding attributes and uses these attributes to query the
+ * decision API.
+ *
+ * This is native JavaScript, no JQuery. It uses the API JSONP interface to get
+ * around CORS and related issues. A script is added with a callback on
+ * `window`. The promise is rejected if there are errors with the request or the
+ * response doesn't look correct.
+ *
+ * Currently, only two parameters are supported with the ad placement: publisher
+ * id and the place type. All of this is determined by the server and this
+ * client so far only renders the API return HTML.
+ *
+ * This can be loaded async. CSS styles are preloaded via webpack
+ * `to-string-loader`.
+ *
+ * Usage:
+ *
+ *     <script async src="ethicalads.min.js"></script>
+ *     <div data-ethical-ad-publisher="foo" data-ethical-ad-type="text"></div>
+ */
+
+import * as ad_css from "./styles.scss";
+
+const AD_DECISION_URL = "https://server.ethicalads.io/api/v1/decision/";
+
+/* Placement object to query decision API and return an Element node
+ *
+ * @param publisher
+ * @param ad_type
+ */
+export class Placement {
+  constructor(publisher, ad_type = "readthedocs-sidebar") {
+    this.publisher = publisher;
+    this.ad_type = ad_type;
+  }
+
+  /* Load placement from decision API and create DOM element
+   *
+   * @returns Promise
+   */
+  load() {
+    const id = "ad_" + Date.now();
+    const url_params = new URLSearchParams({
+      publisher: this.publisher,
+      ad_types: this.ad_type,
+      div_ids: id,
+      callback: id,
+      format: "jsonp",
+    });
+    const url = new URL(AD_DECISION_URL + "?" + url_params.toString());
+    const promise = new Promise((resolve, reject) => {
+      window[id] = (response) => {
+        if (response && response.id) {
+          return resolve(response);
+        } else {
+          return reject(
+            new Error("Placement is configured with invalid parameters.")
+          );
+        }
+      };
+
+      var script = document.createElement("script");
+      script.src = url;
+      script.type = "text/javascript";
+      script.async = true;
+      script.addEventListener("error", reject);
+      document.getElementsByTagName("head")[0].appendChild(script);
+    }).then((response) => {
+      const element = document.createElement("div", { id: this.id });
+      element.innerHTML = response.html;
+
+      var impression_view = document.createElement("img", {
+        src: response.view_url,
+        style: "display: none;",
+      });
+      element.appendChild(impression_view);
```

Comment:
Just a callout here and no action is required. This will immediately cause this image to be loaded regardless of whether the ad is positioned in the viewport. We will need to have agreements with our publishers to ensure that the ad is visible and above the fold.

Reply:
This was just copied from the example js file, I've removed it entirely. Perhaps worth coming back to with something that detects the image in the viewport?